clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -2271,6 +2303,18 @@ private StoreSetting getSettingsFromController(ControllerClient controllerClient
     storeSetting.isWriteComputeEnabled = storeResponse.getStore().isWriteComputationEnabled();
     storeSetting.isIncrementalPushEnabled = storeResponse.getStore().isIncrementalPushEnabled();
     storeSetting.storeRewindTimeInSeconds = DEFAULT_RE_PUSH_REWIND_IN_SECONDS_OVERRIDE;
+    if (setting.isTargetedRegionPush && setting.targetedRegions == null) {
+      // only override the targeted regions if it is not set and it is a single region push
+      setting.targetedRegions = storeResponse.getStore().getNativeReplicationSourceFabric();
+      if (setting.targetedRegions == null) {
```

Comment:
I see getNativeReplicationSourceFabric default value is empty string, regardless good to use `StringUtils.isEmpty` here

Reply:
That's my miss.  Thanks for catching. Let me update it

Reply:
Good. can we add a test to catch this?

Reply:
done